t/unit/worker/test_request.py

```diff
@@ -449,6 +453,23 @@ def test_terminate__task_started(self):
             job.terminate(pool, signal='TERM')
             pool.terminate_job.assert_called_with(job.worker_pid, signum)
 
+    def test_cancel__pool_ref(self):
+        pool = Mock()
+        signum = signal.SIGTERM
+        job = self.get_request(self.mytask.s(1, f='x'))
+        job._apply_result = Mock(name='_apply_result')
+        with self.assert_signal_called(
+                task_retry, sender=job.task, request=job._context,
+                einfo=None):
+            job.time_start = monotonic()
+            job.worker_pid = 314
+            job.cancel(pool, signal='TERM')
+            job._apply_result().terminate.assert_called_with(signum)
+
+            job._apply_result = Mock(name='_apply_result2')
```

Comment:
Does this need an `assert_called_once_with()` as well?

Reply:
The original test didn't include it. Feel free to do so, if you wish this to be fixed.
We can open an issue about it but I think it's rather minor.